tests/unit/utils/test_utils_tei.py

```diff
@@ -0,0 +1,207 @@
+"""Unit tests for the TEI exporter."""
+
+import mock
+import pytest
+
+from inspirehep.modules.converttohal import tei
+from invenio_records.api import Record
+
+
+def test_parse_structures_no_authors():
+    no_authors = Record({})
+
+    expected = []
+    result = tei._parse_structures(no_authors)
+
+    assert expected == result
+
+
+def test_parse_structures_one_author_no_affiliations():
+    one_author_no_affiliations = Record({
+        'authors': [{}]
+    })
+
+    expected = []
+    result = tei._parse_structures(one_author_no_affiliations)
+
+    assert expected == result
+
+
+@mock.patch('inspirehep.modules.converttohal.tei.replace_refs')
+def test_parse_structures_one_author_invalid_affiliation_record(r_r):
+    r_r.return_value = {'record': None}
+
+    one_author_invalid_affiliation_record = Record({
+        'authors': [
+            {'affiliations':
```

Comment:
This looks still pending: formatting is not very consistent. Are you using something to do it automatically?